src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,281 @@
+    # Sort intersection points by insertion order in b_list
+    sort!(a_idx_list, by = x-> a_list[x].neighbor + a_list[x].fracs[2])
+    # Initialize needed values and lists
+    n_b_edges = _nedge(poly_b)
+    n_intr_pts = length(a_idx_list)
+    b_list = Vector{PolyNode{T}}(undef, n_b_edges + n_intr_pts)
+    intr_curr = 1
+    b_count = 0
+    # Loop over points in poly_b and add each point and intersection point
+    for (i, p) in enumerate(GI.getpoint(poly_b))
+        (i == n_b_edges + 1) && break
+        b_count += 1
+        pt = (T(GI.x(p)), T(GI.y(p)))
+        b_list[b_count] = PolyNode(0, pt, false, 0, false, (zero(T), zero(T)))
+        if intr_curr ≤ n_intr_pts
+            curr_idx = a_idx_list[intr_curr]
+            curr_node = a_list[curr_idx]
+            while curr_node.neighbor == i  # Add all intersection points in current edge
+                b_count += 1
+                b_list[b_count] = PolyNode(curr_node.idx, curr_node.point, true, curr_idx, false, curr_node.fracs)
+                curr_node.neighbor = b_count
+                intr_curr += 1
+                intr_curr > n_intr_pts && break
+                curr_idx = a_idx_list[intr_curr]
+                curr_node = a_list[curr_idx]
+            end
+        end
+    end
+    sort!(a_idx_list)  # return a_idx_list to order of points in a_list
+    return b_list
+end
+
+#=
+    _flag_ent_exit(poly_b, a_list)
+
+This function flags all the intersection points as either an 'entry' or 'exit' point in
+relation to the given polygon.
+=#
+function _flag_ent_exit!(poly, pt_list)
+    local status
+    for ii in eachindex(pt_list)
+        if ii == 1
+            status = !_point_filled_curve_orientation(
+                pt_list[ii].point, poly;
+                in = true, on = false, out = false
+            )
+        elseif pt_list[ii].inter
+            pt_list[ii].ent_exit = status
+            status = !status
+        end
+    end
+    return
+end
+
+#=
+    _trace_polynodes(a_list, b_list, a_idx_list, f_step)::Vector{GI.Polygon}
+
+This function takes the outputs of _build_ab_list and traces the lists to determine which
+polygons are formed as described in Greiner and Hormann. The function f_step determines in
+which direction the lists are traced.  This function is different for intersection,
+difference, and union. f_step must take in two arguments: the most recent intersection
+node's entry/exit status and a boolean that is true if we are currently tracing a_list and
+false if we are tracing b_list. The functions used for each clipping operation are follows:
+    - Intersection: (x, y) -> x ? 1 : (-1)
+    - Difference: (x, y) -> (x ⊻ y) ? 1 : (-1)
+    - Union: (x, y) -> (x ⊻ y) ? 1 : (-1)
+
+A list of GeoInterface polygons is returned from this function. 
+=#
+function _trace_polynodes(a_list, b_list, a_idx_list, f_step)
+    n_a_pts, n_b_pts = length(a_list), length(b_list)
+    n_intr_pts = length(a_idx_list)
+    return_polys = Vector{GI.Polygon}(undef, 0)
```

Comment:
Oh! How would you suggest getting around this? Making a vector with the first item as an object and then pushing? I don't understand the `GI.Polygon` type parameters...

Reply:
Sorry I made them pretty complicated. First two are hasz/is3d and hasm (for measures). Then it can wrap literally anything that is geointerface compatible or a nested vector of points

I guess construct one to get the type

Reply:
This doesn't quite work since we still need a type even if there aren't any polygons made. I hard-coded it for now and maybe we can add options for flexibility there later. But I would like to get this preliminarily merged before we dive too deep into that.